sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/InvocationVisitor.cs

```diff
@@ -148,9 +155,18 @@ private ProgramState HandleInstanceEqualsCall()
             return SetConstraintOnValueEquals(equals, newProgramState);
         }
 
-        private static readonly ImmutableHashSet<string> IsNullMethodNames = ImmutableHashSet.Create(
+        private ProgramState HandleNullValidationMethod(int validatedArgumentIndex, int invocationArgsCount) =>
+            programState
+                .PopValues(invocationArgsCount - validatedArgumentIndex - 1)
+                .PopValue(out var guardedArgumentValue)
+                .PopValues(validatedArgumentIndex)
+                .SetConstraint(guardedArgumentValue, ObjectConstraint.NotNull);

```

Comment:
Set NotNull on the argument that has ValidatedNotNull attribute, ignore the rest